sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitApplicationAppeal.e2e-spec.ts

```diff
@@ -732,6 +734,224 @@ describe("StudentAppealStudentsController(e2e)-submitApplicationAppeal", () => {
+        {
+          initialValues: {
+            supportingUserType: SupportingUserType.Parent,
+            fullName: "Parent Two",
+          },
+        },
+      );
+      await db.supportingUser.save([parent1, parent2]);
+
+      // Part of the payload data that will be re-populated by the server.
+      // This re-populated data is provided to form.io to execute dry run submission.
+      const payloadDataToBeRePopulatedByServer = {
+        parents: [
+          { id: parent1.id, fullName: parent1.fullName },
+          // Expect the API to re-populate the parent details with correct id and name.
+          { id: 1, fullName: "Some manipulated name" },
+        ],
+      };
+      // Create a temporary file for step parent waiver appeal.
+      const stepParentWaiverSupportingFile = await saveFakeStudentFileUpload(
+        appDataSource,
+        {
+          student,
+          creator: student.user,
+        },
+        { fileOrigin: FileOriginType.Temporary },
+      );
+      // Prepare the data to submit step parent waiver appeal.
+      const stepParentWaiverAppealData = {
+        ...payloadDataToBeRePopulatedByServer,
+        selectedParent: parent1.id,
+        stepParentWaiverSupportingDocuments: [
+          {
+            url: `student/files/${stepParentWaiverSupportingFile.uniqueFileName}`,
+            hash: "",
+            name: stepParentWaiverSupportingFile.uniqueFileName,
+            size: 4,
+            type: "text/plain",
+            storage: "url",
+            originalName: stepParentWaiverSupportingFile.fileName,
+          },
+        ],
+      };
+      const payload: StudentApplicationAppealAPIInDTO = {
+        studentAppealRequests: [
+          {
+            formName: FormNames.StepParentWaiverAppeal,
+            formData: stepParentWaiverAppealData,
+            files: [stepParentWaiverSupportingFile.uniqueFileName],
+          },
+        ],
+      };
+      // Expected submitted data after the server re-populates the correct parent details.
+      const expectedSubmittedData = {
+        ...stepParentWaiverAppealData,
+        parents: [
+          { id: parent1.id, fullName: parent1.fullName },
+          { id: parent2.id, fullName: parent2.fullName },
+        ],
+      };
+      // Mock JWT user to return the saved student from token.
+      await mockJWTUserInfo(appModule, student.user);
+      // Get any student user token.
+      const studentToken = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+      // Mock the form service to validate the dry-run submission result.
+      // and this mock must be removed.
+      const formService = await getProviderInstanceForModule(
+        appModule,
+        AppStudentsModule,
+        FormService,
+      );
+      const dryRunSubmissionMock = jest.fn().mockResolvedValue({
+        valid: true,
+        formName: FormNames.StepParentWaiverAppeal,
+        data: { data: expectedSubmittedData },
```

Comment:
I am not able to follow the reason why the `expectedSubmittedData` should be provided as a result of the mock. I believe that we must let the server-side override these values and assert it (as it happens at line 879).
Providing the expected data as a mock result does not seem needed.